pkg/api/api.go

```diff
@@ -41,6 +41,7 @@ import (
 
 // DistributorPushWrapper wraps around a push. It is similar to middleware.Interface.
 type DistributorPushWrapper func(next push.Func) push.Func
+type ConfigHandlerProvider func(cfg interface{}) http.HandlerFunc
```

Comment:
I'm curious why this isn't the same as function signature as `DefaultConfigHandler`? I would assume if a client wanted to implement a custom version, they might want to replicate the behavior of the old `configHandler` (and customize).

Reply:
You're totally right, it would be better to have one and the same interface.